test/e2e/multinamespace/multinamespace_test.go

```diff
@@ -0,0 +1,142 @@
+package multinamespace
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	ispnv1 "github.com/infinispan/infinispan-operator/pkg/apis/infinispan/v1"
+	cconsts "github.com/infinispan/infinispan-operator/pkg/controller/constants"
+	ispnctrl "github.com/infinispan/infinispan-operator/pkg/controller/infinispan"
+	ispn "github.com/infinispan/infinispan-operator/pkg/infinispan"
+	tconst "github.com/infinispan/infinispan-operator/test/e2e/constants"
+	k8s "github.com/infinispan/infinispan-operator/test/e2e/k8s"
+	tutils "github.com/infinispan/infinispan-operator/test/e2e/utils"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var testKube = k8s.NewTestKubernetes(os.Getenv("TESTING_CONTEXT"))
+var serviceAccountKube = k8s.NewTestKubernetes("")
+
+var log = logf.Log.WithName("main_test")
+
+var MinimalSpec = ispnv1.Infinispan{
+	TypeMeta: tconst.InfinispanTypeMeta,
+	ObjectMeta: metav1.ObjectMeta{
+		Name: tconst.DefaultClusterName,
+	},
+	Spec: ispnv1.InfinispanSpec{
+		Replicas: 1,
+	},
+}
+
+func TestMain(m *testing.M) {
+	nsAsString := strings.ToLower(tconst.MultiNamespace)
+	namespaces := strings.Split(nsAsString, ",")
+	if "TRUE" == tconst.RunLocalOperator {
+		for _, namespace := range namespaces {
+			testKube.DeleteNamespace(namespace)
+		}
+		testKube.DeleteCRD("infinispans.infinispan.org")
+		testKube.DeleteCRD("caches.infinispan.org")
+		testKube.DeleteCRD("backup.infinispan.org")
+		testKube.DeleteCRD("restore.infinispan.org")
+		for _, namespace := range namespaces {
+			testKube.NewNamespace(namespace)
+		}
+		stopCh := testKube.RunOperator(nsAsString, "../../../deploy/crds/")
+		code := m.Run()
+		close(stopCh)
+		os.Exit(code)
+	} else {
+		code := m.Run()
+		os.Exit(code)
+	}
+}
+
+// Test if single node working correctly
+func TestMultinamespaceNodeStartup(t *testing.T) {
+	// Create a resource without passing any config
+	nsAsString := strings.ToLower(tconst.MultiNamespace)
+	namespaces := strings.Split(nsAsString, ",")
+	var wg sync.WaitGroup
+	for _, namespace := range namespaces {
+		spec := MinimalSpec.DeepCopy()
+		spec.Namespace = namespace
+		// Register it
+		testKube.CreateInfinispan(spec, namespace)
+		defer testKube.DeleteInfinispan(spec, tconst.SinglePodTimeout)
+		wg.Add(1)
+		go func() {
+			waitForPodsOrFail(spec, 1)
+			wg.Done()
+		}()
+	}
+	wg.Wait()
+}
+
+func waitForPodsOrFail(spec *ispnv1.Infinispan, num int) {
```

Comment:
All of the functions below this appear to be the same as in `main_test.go`. We should centralise these in a single .go file and import them in `main_test.go` and `multinamespace_test.go`.

Reply:
I tried to centralize those methods in one of the utils package but I got problems with cycling deps.
But yeah probably I can leave them in the test package in separated file. let me try

Reply:
@ryanemerson methods also rely on package variable `testKube` so I need to change method signature and all the calls around.
Since it's not as simple as moving methods, I would prefer to refact and resolve the cyclic deps in a separate pr.